### PR TITLE
deps(requirments): bump `atc-router` to `v1.4.0`

### PR DESCRIPTION
**Pull request by dndx**
Changelog: https://github.com/Kong/atc-router/releases/tag/v1.4.0

[KAG-3403](https://konghq.atlassian.net/browse/KAG-3403)

### CODE REVIEW

**Comment by chronolaw** [comment]
changelog  required.

**Comment by chronolaw** [comment]
And `Cargo.toml` is still `1.3.1` , should we bump it too?

File: changelog/unreleased/kong/bump-atc-router.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+message: Bumped atc-router from 1.2.0 to 1.4.0
```
Comment (chronolaw) [inline changelog/unreleased/kong/bump-atc-router.yml:1]:
```suggestion
message: Bumped atc-router from 1.3.1 to 1.4.0
```

Comment (dndx) [reply changelog/unreleased/kong/bump-atc-router.yml:1]: 1.3.1 was unreleased, we should use the last released version here which is `1.2.0`:

https://github.com/Kong/kong/blob/3.5.0/.requirements

**Review by ADD-SP — APPROVED** [review]

**Review by chronolaw — APPROVED** [review]

**Pull request merged**